### PR TITLE
Added Structural types and View Bound examples

### PR DESCRIPTION
**Pull request by sibisasidharan**

### CODE REVIEW

**Comment by sujeshchirackkal** [comment]
http://stackoverflow.com/questions/4465948/what-are-scala-context-and-view-bounds this says viewbounds are deprecated and should not be used. Did you happen to use this anywhere

**Comment by sibisasidharan** [comment]
@sujeshchirackkal I didn't use this anywhere, was looking into `Structural Types` when I found this on https://twitter.github.io/scala_school/advanced-types.html. I didn't get any proper deprecation warnings.   

In that case, I think the view bounds example will not be useful, will try adding some example for `Context bound`.

**Comment by sujeshchirackkal** [comment]
Thats fine. keep this anyway and add context bound as well. maybe you can keep the reference to the link which i commented. after all it is a learning repo. so we can keep this example. 👍

**Comment by sibisasidharan** [comment]
@sujeshchirackkal I have added those references and context bound example .

File: src/main/scala/com/training/scala/getstarted/types/ContextBounds.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.training.scala.getstarted.types
+
+/**
+  * Context Bound is used to specify a implicit value is should be available.
```
Comment (sujeshchirackkal) [inline src/main/scala/com/training/scala/getstarted/types/ContextBounds.scala]: sentence to be corrected.

File: src/main/scala/com/training/scala/getstarted/types/ContextBounds.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  *
+  * Eg:
+  *   `def fun[A: B](a: A) = a.get()`
```
Comment (sujeshchirackkal) [inline src/main/scala/com/training/scala/getstarted/types/ContextBounds.scala]: `def fun[A : B](a: A) = a.get()`, stylistic one. Have seen a space before and after ":"

**Comment by sujeshchirackkal** [comment]
👍 please merge unless no other comments from @vineethMM

**Pull request merged**